src/Infrastructure/Network/NetworkClient.cc

```diff
@@ -59,76 +60,286 @@ Task<Result<UserInfoEntity>> NetworkClient::getUserInfo() {
+    } catch (const nlohmann::json::exception& e) {
+        co_return Err(Error(Error::JsonDes, e.what()));
+    }
+
+    co_return Ok(entity);
+}
+
+Task<Result<EventQueryRes>> NetworkClient::getDepartmentEventList(std::string const& department,
+                                                                  int page,
+                                                                  int size) {
+    auto result = co_await this->request<api::Evento>(http::verb::get,
+                                                      endpoint("/v2/client/event/query",
+                                                               {{"page", std::to_string(page)},
+                                                                {"size", std::to_string(size)},
+                                                                {"larkDepartmentName",
+                                                                 department}}));
+    if (result.isErr())
+        co_return Err(result.unwrapErr());
+
+    EventQueryRes entity;
+    try {
+        nlohmann::from_json(result.unwrap(), entity);
+    } catch (const nlohmann::json::exception& e) {
+        co_return Err(Error(Error::JsonDes, e.what()));
+    }
+
+    co_return Ok(entity);
+}
+
+Task<Result<EventQueryRes>> NetworkClient::getEventList(std::initializer_list<urls::param> params) {
+    auto result = co_await this->request<api::Evento>(http::verb::get,
+                                                      endpoint("/v2/client/event/query", params));
+    if (result.isErr())
+        co_return Err(result.unwrapErr());
+
+    EventQueryRes entity;
+    try {
+        nlohmann::from_json(result.unwrap(), entity);
+    } catch (const nlohmann::json::exception& e) {
+        co_return Err(Error(Error::JsonDes, e.what()));
+    }
+
+    co_return Ok(entity);
 }
 
 Task<Result<AttachmentEntity>> NetworkClient::getAttachment(int eventId) {
-    // TODO: implement me!
-    co_return Error(Error::Unknown);
+    auto result = co_await this->request<api::Evento>(
+        http::verb::get, endpoint(std::format("api/v2/client/event/{}/attachments", eventId)));
+    if (result.isErr())
+        co_return Err(result.unwrapErr());
+
+    AttachmentEntity entity;
+    try {
+        nlohmann::from_json(result.unwrap(), entity);
+    } catch (const nlohmann::json::exception& e) {
+        co_return Err(Error(Error::JsonDes, e.what()));
+    }
+
+    co_return Ok(entity);
 }
 
 Task<Result<FeedbackEntity>> NetworkClient::getUserFeedback(int eventId) {
-    // TODO: implement me!
-    co_return Error(Error::Unknown);
+    auto result = co_await this
+                      ->request<api::Evento>(http::verb::get,
+                                             endpoint(std::format("/v2/client/event/{}/feedback",
+                                                                  eventId)));
+    if (result.isErr())
+        co_return Err(result.unwrapErr());
+
+    FeedbackEntity entity;
```

Comment:
```suggestion
    std::optional<FeedbackEntity> entity;
```
需要考虑到 如果用户没有反馈则返回 `null` 的情况